CollectiveFocus/CFM_Backend
# [CFM_58] API to upload an image /v1/image

## Changes
- Add `POST /v1/image` route handler.
- Add `S3Service` adapter class for interfacing with S3.

## Testing
1. Start localstack container:
    ```
    docker compose up
    ```
1. Start local SAM API
    ```
    sam local start-api --parameter-overrides ParameterKey=Environment,ParameterValue=local --docker-network cfm-network
    ```
1. Upload image (replace `<file-path>` with your actual image path like `"@/home/user/Downloads/sample.webp"`)
    ```
    curl --request POST \
      --url http://localhost:3000/v1/image \
      --header 'Content-Type: image/webp' \
      --data-binary "@<file-path>"
    ```
1. Check that it returns an URL and you can open the URL from your browser. Sample response:
    ```
    {"photoURL": "http://localhost:4566/fridge-report/dae7ad47-f1d5-4dc1-9214-03b2a68a6cb9.webp"}
    ```
